examples/statistics/confidence_ellipse.py

```diff
@@ -0,0 +1,222 @@
+# Note that the shape for the weak correlation (right) is an ellipse,
+# not a circle because x and y are differently scaled.
+# However, the fact that x and y are uncorrelated is shown by
+# the axes of the ellipse being aligned with the x- and y-axis
+# of the coordinate system.
+
+np.random.seed(0)
+
+PARAMETERS = {
+    'Positive correlation': np.array([[0.85, 0.35],
+                                      [0.15, -0.65]]),
+    'Negative correlation': np.array([[0.9, -0.4],
+                                      [0.1, -0.6]]),
+    'Weak correlation': np.array([[1, 0],
+                                  [0, 1]]),
+}
+
+mu = 2, 4
+scale = np.array([3, 5]).T
+
+fig, axs = plt.subplots(1, 3, figsize=(9, 3))
+for ax, (title, dependency) in zip(axs, PARAMETERS.items()):
+    x, y = get_correlated_dataset(800, dependency, mu, scale)
+    ax.scatter(x, y, s=0.5)
+
+    ax.axvline(c='grey', lw=1)
+    ax.axhline(c='grey', lw=1)
+
+    confidence_ellipse(x, y, ax, facecolor='none', edgecolor='red')
+
+    ax.scatter(mu[0], mu[1], c='red', s=3)
+    ax.set_title(title)
+
+plt.show()
+
+
+#############################################################################
+#
+# Different number of standard deviations
+# """""""""""""""""""""""""""""""""""""""
+#
+# A plot with n_std = 3 (gray), 2 (blue) and 1 (red)
+
+fig, ax_nstd = plt.subplots(figsize=(6, 6))
+
+dependency_nstd = np.array([
+    [0.8, 0.75],
+    [-0.2, 0.35]
+])
+mu = np.array([0, 0]).T
+scale = np.array([8, 5]).T
+
+ax_nstd.axvline(c='grey', lw=1)
+ax_nstd.axhline(c='grey', lw=1)
+
+x, y = get_correlated_dataset(500, dependency_nstd, mu, scale)
+ax_nstd.scatter(x, y, s=0.5)
+
+confidence_ellipse(x, y, ax_nstd, n_std=1,
+    label=r'$1\sigma$', edgecolor='firebrick')
+confidence_ellipse(x, y, ax_nstd, n_std=2,
+    label=r'$2\sigma$', edgecolor='fuchsia', linestyle='--')
+confidence_ellipse(x, y, ax_nstd, n_std=3,
+    label=r'$3\sigma$', edgecolor='blue', linestyle=':')
+
+ax_nstd.scatter(mu[0], mu[1], c='red', s=3)
+ax_nstd.set_title(f'Different standard deviations')
+ax_nstd.legend()
+plt.show()
+
+
+#############################################################################
+#
+# Using the keyword arguments
+# """""""""""""""""""""""""""
+#
+# Use the kwargs specified for matplotlib.patches.Patch in order
+# to have the ellipse rendered in different ways.
+
+fig, ax_kwargs = plt.subplots(figsize=(6, 6))
+dependency_kwargs = np.array([
+    [-0.8, 0.5],
+    [-0.2, 0.5]
+])
+mu = np.array([2, -3]).T
+scale = np.array([6, 5]).T
+
+ax_kwargs.axvline(c='grey', lw=1)
+ax_kwargs.axhline(c='grey', lw=1)
+
+x, y = get_correlated_dataset(500, dependency_kwargs, mu, scale)
+# Plot the ellipse with zorder=0 in order to demonstrate
+# its transparency (caused by the use of alpha).
+confidence_ellipse(x, y, ax_kwargs,
+    alpha=0.5, facecolor='pink', edgecolor='purple', zorder=0)
+
+ax_kwargs.scatter(x, y, s=0.5)
+ax_kwargs.scatter(mu[0], mu[1], c='red', s=3)
+ax_kwargs.set_title(f'Using kwargs')
```

Comment:
```suggestion
ax_kwargs.set_title('Using kwargs')
```

Probably slipped in through my code example. 🙄 Not needed.